test/tests.py

```diff
@@ -0,0 +1,89 @@
+from representatives_votes.contrib.parltrack import import_dossiers
+from representatives_votes.contrib.parltrack import import_single_dossier
+from representatives_votes.contrib.parltrack import import_votes
+from representatives_votes.models import Dossier, Proposal, Vote
+
+from .parltrack_stub import TestDataSource
+
+
+class ImportTest(TestCase):
+    def setUp(self):
+        for model in (Representative, Dossier, Proposal, Vote):
+            model.objects.all().delete()
+
+    def _get_datapath(self, file):
+        return os.path.join(os.path.dirname(__file__), 'data', file)
+
+    def _test_import(self, scenario, callback):
+        fixture = self._get_datapath('%s_fixture.json' % scenario)
+        expected = self._get_datapath('%s_expected.json' % scenario)
+
+        # Disable django auto fields
+        exclude = ('id', '_state', 'created', 'updated')
+
+        with open(fixture, 'r') as f:
+            callback(f)
+
+        with open(expected, 'r') as f:
+            for obj in Deserializer(f.read()):
+                compare = copy.copy(obj.object.__dict__)
+
+                for f in exclude:
+                    if f in compare:
+                        compare.pop(f)
+
+                type(obj.object).objects.get(**compare)
+
+    def test_parltrack_import_dossiers(self):
+        with self.assertNumQueries(15):
+            self._test_import('dossiers', import_dossiers.main)
+
+    def test_parltrack_import_single_dossier(self):
+        call_command('loaddata', os.path.join(os.path.abspath(
+            parltrack.__path__[0]), 'tests', 'representatives_expected.json'))
+
+        with self.assertNumQueries(16):
+            self._test_import('single', import_single_dossier.main)
+
+    def test_parltrack_import_votes(self):
+        call_command('loaddata', os.path.join(os.path.abspath(
+            parltrack.__path__[0]), 'tests', 'representatives_expected.json'))
+        call_command('loaddata', self._get_datapath('dossiers_expected.json'))
+
+        with self.assertNumQueries(65):
+            self._test_import('votes', import_votes.main)
+
+    def test_dossier_sync(self):
+        call_command('loaddata', self._get_datapath('sync_initial.json'))
+
+        dossier = Dossier.objects.get(reference='2012/2002(INI)')
+
+        with self.assertNumQueries(3):
+            dossier.sync(True, TestDataSource())
+
+        exp = "Agenda for change: the future of EU development policy"
+        assert dossier.title == exp
+
+    def test_dossier_sync_norefresh(self):
+        call_command('loaddata', self._get_datapath('sync_initial.json'))
+
+        dossier = Dossier.objects.get(reference='2012/2002(INI)')
+
+        with self.assertNumQueries(2):
+            dossier.sync(False, TestDataSource())
+
+        assert dossier.title == 'initial title'
+
+        dossier.refresh_from_db()
+
+        exp = "Agenda for change: the future of EU development policy"
+        assert dossier.title == exp
```

Comment:
If you're not using py.test, then assertEquals() and other xUnit-family assertion commands.

For example:

```
import unittest


class SomeTest(unittest.TestCase):
    a = True
    b = False

    def test_assert(self):
        assert self.a == self.b

    def test_xunit_assert(self):
        self.assertEqual(self.a, self.b)
```

With the unittest module, we just get an AssertionError which is not descriptive test_assert:

```
$ python -m unittest test
FF
======================================================================
FAIL: test_assert (test.SomeTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test.py", line 9, in test_assert
    assert self.a == self.b
AssertionError

======================================================================
FAIL: test_xunit_assert (test.SomeTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test.py", line 12, in test_xunit_assert
    self.assertEqual(self.a, self.b)
AssertionError: True != False

----------------------------------------------------------------------
Ran 2 tests in 0.000s

FAILED (failures=2)
```

With py.test, we get descriptive output for both:

```
[env] 03/01 2016 21:17:44 jpic@lue ~ 
$ py.test test.py
==================================== test session starts ====================================
platform linux2 -- Python 2.7.11, pytest-2.8.2, py-1.4.30, pluggy-0.3.1
rootdir: /home/jpic, inifile: 
plugins: django-2.8.0
collected 2 items 

test.py FF

========================================= FAILURES ==========================================
___________________________________ SomeTest.test_assert ____________________________________

self = <test.SomeTest testMethod=test_assert>

    def test_assert(self):
>       assert self.a == self.b
E       AssertionError: assert True == <test.SomeTest testMethod=test_assert>.b
E        +  where True = <test.SomeTest testMethod=test_assert>.a

test.py:9: AssertionError
________________________________ SomeTest.test_xunit_assert _________________________________

self = <test.SomeTest testMethod=test_xunit_assert>

    def test_xunit_assert(self):
>       self.assertEqual(self.a, self.b)
E       AssertionError: True != False

test.py:12: AssertionError
================================= 2 failed in 0.02 seconds ==================================
```

I think that's one of the reasons py.test is hype nowadays ^^